src/modules/rppt_tensor_augmentations.cpp

```diff
@@ -1180,3 +1180,171 @@ rppt_gridmask_host(RppPtr_t srcPtr,
 
     return RPP_SUCCESS;
 }
+
+/******************** spatter ********************/
+
+RppStatus
+rppt_spatter_gpu(RppPtr_t srcPtr,
+                 RpptDescPtr srcDescPtr,
+                 RppPtr_t dstPtr,
+                 RpptDescPtr dstDescPtr,
+                 RpptRGB spatterColor,
+                 RpptROIPtr roiTensorPtrSrc,
+                 RpptRoiType roiType,
+                 rppHandle_t rppHandle)
+{
+#ifdef HIP_COMPILE
+    RpptROI roiTensorPtrSrcHost[dstDescPtr->n];
+    hipMemcpy(roiTensorPtrSrcHost, roiTensorPtrSrc, dstDescPtr->n * sizeof(RpptROI), hipMemcpyDeviceToHost);
+    if (roiType == RpptRoiType::XYWH)
+    {
+        for(int i = 0; i < dstDescPtr->n; i++)
+            if ((roiTensorPtrSrcHost[i].xywhROI.roiWidth > SPATTER_MAX_WIDTH) || (roiTensorPtrSrcHost[i].xywhROI.roiHeight > SPATTER_MAX_HEIGHT))
+                return RPP_ERROR_HIGH_SRC_DIMENSION;
+    }
+    else if (roiType == RpptRoiType::LTRB)
+    {
+        for(int i = 0; i < dstDescPtr->n; i++)
+            if ((roiTensorPtrSrcHost[i].ltrbROI.rb.x - roiTensorPtrSrcHost[i].ltrbROI.lt.x > SPATTER_MAX_XDIM) || (roiTensorPtrSrcHost[i].ltrbROI.rb.y - roiTensorPtrSrcHost[i].ltrbROI.lt.y > SPATTER_MAX_YDIM))
+                return RPP_ERROR_HIGH_SRC_DIMENSION;
+    }
+
+    std::random_device rd;  // Random number engine seed
+    std::mt19937 gen(rd()); // Seeding rd() to fast mersenne twister engine
+    uint2 maskLocArrHost[dstDescPtr->n];
+    for(int i = 0; i < dstDescPtr->n; i++)
+    {
+        std::uniform_int_distribution<> distribX(0, SPATTER_MAX_WIDTH - roiTensorPtrSrcHost[i].xywhROI.roiWidth);
+        std::uniform_int_distribution<> distribY(0, SPATTER_MAX_HEIGHT - roiTensorPtrSrcHost[i].xywhROI.roiHeight);
+        maskLocArrHost[i].x = distribX(gen);
+        maskLocArrHost[i].y = distribY(gen);
+    }
+    uint2 *maskLocArr;
+    hipMalloc(&maskLocArr, dstDescPtr->n * sizeof(uint2));
```

Comment:
same comment as above. please remove allocation from this function.

Reply:
Done